GDScript: template fix for space intent

RameshRavone opened the pull request.

bd339 committed:
Show 3D raycasts when debugging collisions

Represent a raycast by a line segment using the direction and magnitude of the cast_to vector.
The entire line segment will become red while a collision occurs.

tagcup committed:
Added various functions basic math classes. Also enabled math checks only for debug builds.

Added set_scale, set_rotation_euler, set_rotation_axis_angle. Addresses #2565 directly.
Added an euler angle constructor for Basis in GDScript and also exposed is_normalized for vectors and quaternions.
Various other changes mostly cosmetic in nature.

volzhs committed:
Prevent to uncheck selected item with button group

20kdc committed:
Ported/fixed move_and_slide to KinematicBody (3D+floor/ceiling args)

This version of the commit has the on_ceiling/on_wall fix without any
 explaination of that code, since apparently it wasn't complicated enough.

As for the notes at the top of the function, they're still there.

move_and_slide is rather useful for character-controllers, etc.
It reduces the amount of boilerplate code.
Not having move_and_slide makes the APIs somewhat inconsistent.

(It might be nice to figure out a way to share the code between the two
move_and_slide implementations, but that's for someone who knows what
the policy is on shared code like that.)

20kdc committed:
Apply is_ceiling/is_wall swap fix to 2D move_and_slide (minus explaination)

As it turns out, is_ceiling would be true when hitting a wall,
 and is_wall would be true momentarily when hitting a ceiling.

This makes a tiny one-line change to fix that.
Without trying to explain the code for anyone else having to
 mess around with it.

mablin7 committed:
Added the ability to select files as base when creating scripts

mablin7 committed:
Added autocomplete for file paths in the script editor

RameshRavone committed:
Template fix for space intent

RameshRavone committed:
Template fix for space intent

bojidar-bg commented on modules/gdscript/gd_editor.cpp:
```diff
@@ -44,19 +44,23 @@ void GDScriptLanguage::get_string_delimiters(List<String> *p_delimiters) const {
-					   "\t# Called every time the node is added to the scene.\n" +
-					   "\t# Initialization here\n" +
-					   "\tpass\n";
+				    "extends %BASE%\n\n" +
```
Ooooh my. Did you just use both spaces and tabs for aligning there?

I think it might be better if just a single tab (and no spaces) is used for those lines, but I would like to have a confirmation from @akien-mga on that

bojidar-bg replied:
Actually, scratch that, since clang-format would probably fail on both anyway.

So, you should keep clang-format happy too, not just reformat the code :smiley:

RameshRavone replied:
Guess, Its time I drop VIM

akien-mga replied:
Or configure vim :)

RameshRavone committed:
Clang format

RameshRavone committed:
Clang format

bojidar-bg commented:
OSX build fails with:
```
Undefined symbols for architecture armv7:
  "_EDITOR_DEF(String const&, Variant const&)", referenced from:
      GDScriptLanguage::get_template(String const&, String const&) const in libmodules.iphone.debug.arm.a(gd_editor.iphone.debug.arm.o)
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/godot.iphone.debug.arm] Error 1
scons: building terminated because of errors.
```

RameshRavone committed:
Fix for OSX build

RameshRavone committed:
Fix for OSX build

bojidar-bg commented:
Would you mind adding a check for `is_editor_hint` as well, since we don't want to access editor settings in cases it isn't an editor?

RameshRavone commented:
Ok, If I'm updating L51 then I guess I need to update L2409 & L2412

RameshRavone committed:
Enforce access to editor settings

RameshRavone committed:
Enforce access to editor settings

RameshRavone commented:
Can't we check `is_editor_hint` inside _EDITOR_DEF function

akien-mga committed:
Fix property warnings and hide some debug prints

"ALL IS GOOD" was a lie.

In particular, removes verbose "path not recognized" false positive.

The actual logic is to (somewhat naively) check all ResourceFormatLoaders
and to pick the first good match, so no need to warn about the formats
that do not match the type hint.

Paulb23 committed:
Fix cursor position on indent convert

Paulb23 committed:
Fixed convert indent and trim whitespace on save

Commit:
Update snake_case splitting

Commit:
Added a container to EditorNameDialog

karroffel committed:
re-added MultiScript

The very first Godot version (when it was open sourced) had "MultiScript" which lets you use multiple scripts on one object.
With the addition of mulitple new scripting languages (VisualScript, soon C# and GDNative) it can be of use to combine scripts rather than delegating (with huge maintainance cost) or creating child nodes
which could impact performance.

I used the code from 0b806ee as the base and made it work with the current master.

Caitacular committed:
[3.0] Fix bug in Image::_get_pixelb

Fixes issue:
 https://github.com/godotengine/godot/issues/8158

akien-mga committed:
Merge pull request #8502 from karroffel/multiscript

re-added MultiScript

akien-mga committed:
Merge pull request #8496 from akien-mga/stdout-cleanup

Fix property warnings and hide some debug prints

akien-mga committed:
Merge pull request #8102 from bd339/iss3172

Show 3D raycasts when debugging collisions

akien-mga committed:
Merge pull request #8277 from tagcup/math_checks

Added various functions basic math classes. Also enabled math checks …

akien-mga committed:
Merge pull request #8343 from volzhs/prevent_uncheck

Prevent to uncheck selected item with button group

akien-mga committed:
Merge pull request #8395 from 20kdc/master

Ported move_and_slide to KinematicBody (3D)

nunodonato committed:
Fixed #8102, by renaming FixedSpatialMaterial to SpatialMaterial

akien-mga committed:
Merge pull request #8408 from 20kdc/bugfix-kine-mas

Apply is_ceiling/is_wall swap fix to 2D move_and_slide

akien-mga committed:
Merge pull request #8420 from magyar123/pr-script-files-as-base

Added the ability to select files as base when creating scripts

akien-mga committed:
Merge pull request #8444 from magyar123/pr-complete-paths

Script editor now automatically completes file paths in GDScript

akien-mga committed:
Merge pull request #8469 from Melix19/patch-2

Update snake_case splitting

akien-mga commented on modules/gdscript/gd_editor.cpp:
```diff
@@ -2394,16 +2397,18 @@ Error GDScriptLanguage::complete_code(const String &p_code, const String &p_base
 String GDScriptLanguage::_get_indentation() const {
 #ifdef TOOLS_ENABLED
-	bool use_space_indentation = EDITOR_DEF("text_editor/indent/type", "Tabs") == "Tabs" ? 0 : 1;
+	if (OS::get_singleton()->get_main_loop()->cast_to<SceneTree>()->is_editor_hint()) {
```
Do you really need to cast it to access `is_editor_hint()`?

RameshRavone replied:
I suppose we can use `SceneTree::get_singleton()->is_editor_hint();` too

akien-mga committed:
Merge pull request #8499 from Melix19/patch-3

Fix "Save/Delete Layout" text field goes outside of the popup

akien-mga committed:
Merge pull request #8500 from Paulb23/space_indent_fixes

Space indent fixes

akien-mga committed:
Merge pull request #8506 from iam13islucky/patch-1

[3.0] Fix bug in Image::_get_pixelb

akien-mga committed:
Merge pull request #8509 from nunodonato/fixspatialmaterial

Fixed #8102, by renaming FixedSpatialMaterial to SpatialMaterial

RameshRavone committed:
Merge branch 'master' of https://github.com/RameshRavone/godot

RameshRavone commented:
Merge conflicts:
